### PR TITLE
Spline data markers scriptable tool

### PR DESCRIPTION
**Pull request by Fredaly**

### CODE REVIEW

File: Source/SplineFollow/Public/Components/SFSplineMarkers.h [hunk 1]
```diff
@@ -158,14 +158,65 @@ FORCEINLINE UTexture2D * ASFSplineMarkerLevelActor::GetSprite() const
...
 
-USTRUCT( BlueprintType )
+UCLASS( DefaultToInstanced, Blueprintable, BlueprintType, Abstract, EditInlineNew )
+class SPLINEFOLLOW_API USFSplineMarkerObject : public UObject
+{
+    GENERATED_BODY()
+
+protected:
+    UPROPERTY( BlueprintReadOnly, EditAnywhere )
+    UTexture2D * Sprite;
```
Comment (TheEmidee) [inline Source/SplineFollow/Public/Components/SFSplineMarkers.h]: TObjectPtr

File: Source/SplineFollow/Public/Components/SFSplineMarkers.h [hunk 2]
```diff
@@ -158,14 +158,65 @@ FORCEINLINE UTexture2D * ASFSplineMarkerLevelActor::GetSprite() const
...
+};
+
+UCLASS()
+class SPLINEFOLLOW_API USFSplineMarkerObject_Data : public USFSplineMarkerObject
+{
+    GENERATED_BODY()
+
+protected:
+    UPROPERTY( BlueprintReadOnly, EditAnywhere, Instanced )
+    USFSplineMarkerData * Data;
```
Comment (TheEmidee) [inline Source/SplineFollow/Public/Components/SFSplineMarkers.h]: TObjectPtr

File: Source/SplineFollow/Public/Components/SFSplineMarkers.h [hunk 3]
```diff
@@ -227,4 +291,11 @@ struct SPLINEFOLLOW_API FSFSplineMarker_Data : public FSFSplineMarker
 FORCEINLINE UTexture2D * FSFSplineMarker_Data::GetSprite() const
 {
     return Sprite;
-}
\ No newline at end of file
+}
+
+UCLASS( Blueprintable, BlueprintType )
+class USFSplineMarkerType : public UObject
```
Comment (TheEmidee) [inline Source/SplineFollow/Public/Components/SFSplineMarkers.h]: what is it used for?

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/SplineFollow/Public/Components/SFSplineMarkers.h (hunk 1 above)
Comment (Fredaly) [reply Source/SplineFollow/Public/Components/SFSplineMarkers.h]: done

File: Source/SplineFollow/Public/Components/SFSplineMarkers.h (hunk 2 above)
Comment (Fredaly) [reply Source/SplineFollow/Public/Components/SFSplineMarkers.h]: done

File: Source/SplineFollow/Public/Components/SFSplineMarkers.h (hunk 3 above)
Comment (Fredaly) [reply Source/SplineFollow/Public/Components/SFSplineMarkers.h]: removed

**Pull request merged**